### PR TITLE
test: refactor test index test

### PR DESCRIPTION
**Pull request by JoanFM**
@bwanglzu I found some random errors

### CODE REVIEW

**Review by bwanglzu — CHANGES_REQUESTED** [review]

File: tests/unit/test_index.py [hunk 1]
```diff
@@ -138,20 +138,25 @@ def start_client(fl):
         t2.start()
 
 
-def test_index():
+@pytest.mark.parametrize('random_workspace_name', ['JINA_TEST_INDEX'])
```
[the remark this answers is not in the document]
Comment (JoanFM) [reply tests/unit/test_index.py:141]: That the yaml file used declares a workspace depending on an env variable

[the remark this answers is not in the document]
Comment (JoanFM) [reply tests/unit/test_index.py:141]: But yes maybe is a mistake from my side

[the remark this answers is not in the document]
Comment (bwanglzu) [reply tests/unit/test_index.py:141]: maybe we can create a new fixture which can takes a specific workspace name as a parameter?

File: tests/unit/test_index.py [hunk 2]
```diff
@@ -138,20 +138,25 @@ def start_client(fl):
...
+    workspace_path = os.environ['JINA_TEST_INDEX']
     f = Flow().add(uses=os.path.join(cur_dir, 'yaml/test-index.yml'), parallel=3, separated_workspace=True)
     with f:
         f.index(input_fn=random_docs(1000))
 
     for j in range(3):
-        assert os.path.exists(f'test2-{j + 1}/test2.bin')
-        assert os.path.exists(f'test2-{j + 1}/tmp2')
+        path = os.path.join(workspace_path, f'test2-{j + 1}/test2.bin')
+        print(f' path {path}')
```
Comment (nan-wang) [inline tests/unit/test_index.py:150]:
```suggestion
```

File: tests/unit/test_index.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (nan-wang) [reply tests/unit/test_index.py:141]: That would be helpful.

**Review by nan-wang — CHANGES_REQUESTED** [review]
minor changes

**Review by hanxiao — APPROVED** [review]

**Pull request merged**